lib/experimental/fonts-library/class-wp-fonts-library.php

```diff
@@ -0,0 +1,618 @@
+     *
+     * @param array $new_fonts_families The new list of fonts families to replace the current content of the fonts library post.
+     * @return WP_REST_Response|WP_Error The updated fonts library post content wrapped in a WP_REST_Response object.
+     */
+    function update_fonts_library ( $new_fonts_families ) {
+        $post = $this->query_fonts_library();
+        $new_fonts_library = array (
+            'fontFamilies' => $new_fonts_families,
+        );
+        $updated_post_data = array (
+            'ID' => $post->ID,
+            'post_content' => wp_json_encode( $new_fonts_library ),
+        );
+        $updated_post = wp_update_post( $updated_post_data );
+        if ( is_wp_error( $updated_post ) ){
+            return $updated_post;
+        }
+        $new_post = $this->query_fonts_library();
+        if ( is_wp_error( $new_post ) ){
+            return $new_post;
+        }
+        return new WP_REST_Response( $new_post->post_content );
+    }
+
+    /**
+     * Fetches the Google Fonts JSON file.
+     *
+     * Reads the "google-fonts.json" file from the file system and returns its content.
+     *
+     * @return WP_REST_Response|WP_Error The content of the "google-fonts.json" file wrapped in a WP_REST_Response object.
+     */
+    function get_google_fonts () {
+        $file = file_get_contents(
+            path_join ( dirname(__FILE__),"google-fonts.json" )
+        );
+        if ( $file ) {
+            return new WP_REST_Response( json_decode( $file ) );
+        }
+        return new WP_Error(
+            'rest_cant_read_google_fonts',
+            __( 'Error reading Google Fonts JSON file.' ),
+            array( 'status' => 500 )
+        );
+    }
+
+    /**
+     * Deletes a specified font asset file from the fonts directory.
+     *
+     * @param string $src The path of the font asset file to delete.
+     * @return bool Whether the file was deleted.
+     */
+    function delete_asset( $src ) {
+        $filename = basename( $src );
+        $file_path = path_join( $this->wp_fonts_dir, $filename );
+
+        if ( ! file_exists( $file_path ) ) {
+            return false;
+        }
+
+        return wp_delete_file( $file_path );
+    }
+
+    /**
+     * Deletes all font face asset files associated with a given font face.
+     *
+     * @param array $font_face The font face array containing the 'src' attribute with the file path(s) to be deleted.
+     */
+    function delete_font_face_assets ( $font_face ) {
+        if ( is_array( $font_face['src'] ) ) {
+            foreach ( $font_face['src'] as $src ) {
+                $was_assets_removed = $this->delete_asset($src);
+                if ( ! $was_assets_removed ) {
+                    // Bail if any of the assets could not be removed
+                    return false;
+                }
+            }
+            return true;
+        } else {
+            return $this->delete_asset( $font_face['src'] );
+        }
```

Comment:
Let's [avoid nesting and return early](https://github.com/piotrplenik/clean-code-php#avoid-nesting-too-deeply-and-return-early-part-1):
```suggestion
        if ( ! is_array( $font_face['src'] ) ) {
            return $this->delete_asset( $font_face['src'] );
        }
        
        foreach ( $font_face['src'] as $src ) {
            $was_assets_removed = $this->delete_asset($src);
            if ( ! $was_assets_removed ) {
                // Bail if any of the assets could not be removed
                return false;
            }
        }
        return true;
        
```

Reply:
changed here: https://github.com/WordPress/gutenberg/pull/50927/commits/645626f1f8b9206c268a4dbc7f3b05ea25e17ac6